modifications/keyrotationcontroller.cpp

```diff
@@ -0,0 +1,48 @@
+#include "keyrotationcontroller.h"
+
+KeyRotationController::KeyRotationController(QObject *parent) : QObject(parent)
+{
+    pathOnBoard = "APM";
+    keyFileName = "key.pem";
+    pathOnSystem = QStandardPaths::writableLocation(QStandardPaths::TempLocation);
+    uploading = false;
+}
+
+void KeyRotationController::startKeyRotation(QString URL)
+{
+    url = URL;
+    if(!fetchKeyFromDrone()){
+        return;
+    }
+}
+
+bool KeyRotationController::fetchKeyFromDrone()
+{
+
+    FTPManager* ftpManager = qgcApp()->toolbox()->multiVehicleManager()->activeVehicle()->ftpManager();
+    QString a, b;
+    connect(ftpManager, SIGNAL(downloadComplete(const QString& , const QString& )), this, SLOT(downloadComplete(const QString&, const QString&)));
+
+    if(!ftpManager->download(pathOnBoard + "/" + keyFileName, pathOnSystem)){
+        qInfo() << " Cannot Download Key.pem";
```

Comment:
Maybe change this log to a user friendly log like "Unable to download key file to UAS" and add a debug log for qDebug() << "Download to" <<   pathOnBoard + "/" + keyFileName << "failed!"